source/module_hsolver/diagosca.cpp

```diff
@@ -0,0 +1,278 @@
+//=====================
+// AUTHOR : Peize Lin
+// DATE : 2021-11-02
+// REFACTORING AUTHOR : Daye Zheng
+// DATE : 2022-04-14
+//=====================
+
+#include "diagosca.h"
```

Comment:
I guess `sca` for "scalapack" here. This name is not so good - either hard to understand or too long. How about we calling it `diago_blas`?

Reply:
Which name is better? "diago_blas" or "diagoblas"?

Reply:
> Which name is better? "diago_blas" or "diagoblas"?

Haha, this situation is the same as `hamilt_lcao` I mentioned in comments above.
Personally, I think an underscore(or any separator) helps stress the hierarchy: `hamilt` for the base class, `hamilt_lcao`, `hamilt_pw` for derived class.

Reply:
ok